include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -436,6 +463,61 @@ class alignment_algorithm :
         this->check_score_last_column(last_column_view, get<3>(cache));
     }
 
+    template <typename first_range_t, typename second_range_t>
+    auto compute_traceback(first_range_t const & first_range,
+                           second_range_t const & second_range,
+                           alignment_coordinate const & end_coordinate)
+    {
+        using first_seq_value_type = value_type_t<first_range_t>;
+        using second_seq_value_type = value_type_t<second_range_t>;
+
+        // Parse the traceback
+        auto [begin_coordinate, first_gap_segments, second_gap_segments] = this->parse_traceback(end_coordinate);
```

Comment:
Ah, found it. It refers to the first (horizontal) and second (vertical) sequence.
Think again about these names, because you often use directional names, as up/left/horizontal etc. and it should be consistent all over the alignment module.

Reply:
So the entire public API will refer to first and second range as for the pairwise alignment the user is not interested in the internal layout. I kept this naming scheme for most of the internal implementation whenever I refer to the ranges. When I refer to horizontal or vertical then only at the places where the layout of the matrix is relevant or can be easier understood. (AFAICT this is only in comments)
For the trace directions: up, left and diagonal. They were introduced for the edit distance alignment and I think it is ok to give them the name in the way they walk. Like go left, up or diagonal. One could rename them to horizontal and vertical but I don't see any benefit in this. If you think otherwise, please open an issue, so the group can discuss it. It would be another PR anyway.